src/control-plane/backend/lambda/api/model/pipeline.ts

```diff
@@ -518,21 +519,40 @@ export class CPipeline {
     this.pipeline.templateVersion = FULL_SOLUTION_VERSION;
     await this.resourcesCheck();
     this.pipeline.workflow = await generateWorkflow(this.pipeline, this.resources!);
+    this.stackManager.setWorkflow(this.pipeline.workflow);
     this.stackManager.setExecWorkflow(this.pipeline.workflow);
     const oldStackNames = this.stackManager.getWorkflowStacks(oldPipeline.workflow?.Workflow!);
     // update workflow
     this.stackManager.upgradeWorkflow(oldStackNames);
     // update workflow callback
     this.stackManager.setPipelineWorkflowCallback(executionName);
+    // update workflow ingestion stack template
+    await this._updateIngestionServerStackTemplate(oldPipeline);
     // create new execution
     const execWorkflow = this.stackManager.getExecWorkflow();
     const executionArn = await this.stackManager.execute(execWorkflow, executionName);
     this._setExecution(executionArn);
+    this.pipeline.workflow = this.stackManager.getWorkflow();
     this.pipeline.statusType = PipelineStatusType.UPDATING;
     // update pipeline metadata
     await store.updatePipeline(this.pipeline, oldPipeline);
   }
 
+  private async _updateIngestionServerStackTemplate(oldPipeline: IPipeline): Promise<void> {
```

Comment:
It's not right. We still ship the v1 template for upgrading the existing pipeline using v1.